tests/datasource/fluent/test_viral_snippets.py

```diff
@@ -328,5 +332,49 @@ def test_context_add_or_update_datasource(
     assert updated_datasource.connection_string == "sqlite:///"
 
 
+@pytest.fixture
+def random_datasource(seeded_file_context: FileDataContext) -> Datasource:
+    datasource = random.choice(list(seeded_file_context.fluent_datasources.values()))
+    logger.info(f"Random DS - {pf(datasource.dict(), depth=1)}")
+    return datasource
+
+
+def test_sources_delete_removes_datasource_from_yaml(
+    random_datasource: Datasource,
+    seeded_file_context: FileDataContext,
+):
+    print(f"Delete -> '{random_datasource.name}'\n")
+
+    ds_delete_method: Callable[[str], None] = getattr(
+        seeded_file_context.sources, f"delete_{random_datasource.type}"
+    )
+    ds_delete_method(random_datasource.name)
+
+    yaml_path = pathlib.Path(
+        seeded_file_context.root_directory, seeded_file_context.GX_YML
+    ).resolve(strict=True)
+    yaml_contents = YAML.load(yaml_path.read_text())
+    print(f"{pf(yaml_contents, depth=2)}")
+
+    assert random_datasource.name not in yaml_contents["fluent_datasources"]
+
+
+def test_ctx_delete_removes_datasource_from_yaml(
+    random_datasource: Datasource, seeded_file_context: FileDataContext
+):
+    print(f"Delete -> '{random_datasource.name}'\n")
+
+    seeded_file_context.delete_datasource(random_datasource.name)
+    seeded_file_context._save_project_config()
```

Comment:
The normal `context.delete_datasource()` never calls `_save_project_config()` so this extra call is needed to persist the changes to `great_expectations.yml`.

This seems like a weird bifurcation of behavior. See the related comment above.
https://github.com/great-expectations/great_expectations/pull/7843/files#r1187848283